cpp/CMakeLists.txt

```diff
@@ -1,53 +1,117 @@
+#=============================================================================
+# Copyright (c) 2018-2019, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#=============================================================================
 cmake_minimum_required (VERSION 3.12)
 
 project(NVStrings VERSION 0.7.0 LANGUAGES C CXX CUDA)
-find_package(PythonLibs 3 REQUIRED)
 
-option(CMAKE_CXX11_ABI "Enable the GLIBCXX11 ABI" ON)
-option(BUILD_TESTS "Configure CMake to build tests" OFF)
+###################################################################################################
+# - build type ------------------------------------------------------------------------------------
+
+# Set a default build type if none was specified
+set(DEFAULT_BUILD_TYPE "Release")
+
+if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
+  message(STATUS "Setting build type to '${DEFAULT_BUILD_TYPE}' since none specified.")
+  set(CMAKE_BUILD_TYPE "${DEFAULT_BUILD_TYPE}" CACHE
+      STRING "Choose the type of build." FORCE)
+  # Set the possible values of build type for cmake-gui
+  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS
+    "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
+endif()
+
+###################################################################################################
+# - compiler options ------------------------------------------------------------------------------
 
 set(CMAKE_CXX_STANDARD 14)
 set(CMAKE_C_COMPILER $ENV{CC})
 set(CMAKE_CXX_COMPILER $ENV{CXX})
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
 set(CMAKE_CUDA_STANDARD 14)
 set(CMAKE_CUDA_STANDARD_REQUIRED ON)
 
 SET(CMAKE_BUILD_WITH_INSTALL_RPATH TRUE)
 SET(CMAKE_INSTALL_RPATH "\$ORIGIN")
 
 if (WIN32) 
-set (DEFINES  ${DEFINES}
--D"_CRT_SECURE_NO_DEPRECATE"  
--D"_SCL_SECURE_NO_DEPRECATE" 
-)
+    set (DEFINES ${DEFINES}
+         -D"_CRT_SECURE_NO_DEPRECATE"  
+         -D"_SCL_SECURE_NO_DEPRECATE" 
+    )
 endif()
+add_definitions(${DEFINES})
+
+if(CMAKE_COMPILER_IS_GNUCXX)
+    option(CMAKE_CXX11_ABI "Enable the GLIBCXX11 ABI" ON)
+    if(NOT CMAKE_CXX11_ABI)
+        message(STATUS "NVStrings: Disabling the GLIBCXX11 ABI")
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -D_GLIBCXX_USE_CXX11_ABI=0")
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D_GLIBCXX_USE_CXX11_ABI=0")
+        set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -D_GLIBCXX_USE_CXX11_ABI=0")
```

Comment:
Just a question, happy to move this elsewhere as well. When do we plan to move to the new C++ ABI?

Reply:
We already default to the new ABI and the option is left in for legacy if someone needs the old ABI.